src/components/CategoryNavContainer/CategoryNav/CategoryNav.js

```diff
@@ -1,34 +1,62 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { NavLink } from 'react-router-dom';
 import style from './CategoryNav.scss';
 import { NO_CATEGORY } from '../../../constants';
 
-const CategoryNav = props => (
-  <div>
-    <NavLink
-      className={style.link}
-      activeClassName={style.link__active}
-      to={`/${props.cityId}/${NO_CATEGORY}`}
-    >Overview</NavLink>
+class CategoryNav extends Component {
+  componentDidMount() {
+    this.scrollToActiveNavLink();
+  }
 
-    {props.categories.map(category => (
-      <NavLink
-        className={style.link}
-        activeClassName={style.link__active}
-        key={category.id}
-        to={`/${props.cityId}/${category.id}`}
-      > {category.name} </NavLink>
-    ))}
-  </div>
-);
+  componentDidUpdate() {
+    this.scrollToActiveNavLink();
+  }
+
+  scrollToActiveNavLink() {
+    if (this.activeCategoryEl) {
+      this.el.scrollLeft = this.activeCategoryEl.parentElement.offsetLeft - 20;
+    }
+  }
+
+  render() {
+    return (
+      <div ref={(el) => { this.el = el; }} className={style.categoryNav}>
+        <NavLink
+          className={style.link}
+          activeClassName={style.link__active}
+          to={`/${this.props.cityId}/${NO_CATEGORY}`}
+        >
+          <div className={style.overview}>Overview</div>
+        </NavLink>
+
+        {this.props.categories.map(category => (
+          <NavLink
+            className={style.link}
+            activeClassName={style.link__active}
+            key={category.id}
+            to={`/${this.props.cityId}/${category.id}`}
+          >
+            <div
+              className={style[category.colorName]}
+              ref={(el) => {
+                if (this.props.categoryId === category.id) this.activeCategoryEl = el;
+              }}
+            > {category.navName || category.name} </div>
```

Comment:
Minor: My first thought when I see this is that there are spaces for a reason and they must be doing something. It seems funny to add whitespace in the markup on purpose knowing that it will be stripped out by React. It isn't done anywhere else (e.g. `<div className={style.overview}>Overview</div>` above).

Reply:
I've shifted the spaces to the other side of the curly braces :)